protos/odpf/stencil/stencil.proto

```diff
@@ -0,0 +1,85 @@
+syntax = "proto3";
+package odpf.stencil.v1;
+
+import "google/api/annotations.proto";
+import "google/api/field_behavior.proto";
+import "google/protobuf/descriptor.proto";
+import "protoc-gen-openapiv2/options/annotations.proto";
+
+
+option go_package = "github.com/odpf/stencil/server/api/v1/pb";
+
+// These annotations are used when generating the OpenAPI file.
+option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_swagger) = {
+	info: {
+		version: "0.1.3";
+	};
+	external_docs: {
+		description: "Stencil server";
+	}
+	schemes: HTTP;
+};
+
+message Snapshot {
+	int64 id = 1;
+	string namespace = 2 [(google.api.field_behavior) = REQUIRED];
+	string name = 3;
+	string version = 4;
+	bool latest = 5;
+}
+
+message DownloadRequest {
+	string namespace = 1 [(google.api.field_behavior) = REQUIRED];
+	string name = 2 [(google.api.field_behavior) = REQUIRED];
+	string version = 3 [(google.api.field_behavior) = REQUIRED];
+	repeated string fullnames = 4;
+}
+
+message DownloadResponse {
+	bytes data = 1;
+}
+
+message UploadRequest {
+	Snapshot snapshot = 1 [(google.api.field_behavior) = REQUIRED];
+	bytes data = 4 [(google.api.field_behavior) = REQUIRED];
+	bool dryrun = 5;
+	repeated string skiprules = 6;
+}
+
+message UploadResponse {
+	bool success = 1;
+	bool dryrun = 2;
+	string errors = 3;
+}
+
+message SnapshotList {
+	repeated Snapshot snapshots = 1;
+}
+
+message ListSnapshotRequest {
+	string namespace = 1;
+	string name = 2;
+	string version = 3;
+	bool latest = 4;
+}
+
+message UpdateLatestRequest {
+	int64 id = 1;
+	bool latest = 2;
+}
+
+service StencilService {
+	rpc Upload (UploadRequest) returns (UploadResponse) {}
+	rpc Download (DownloadRequest) returns (DownloadResponse) {}
+	rpc List(ListSnapshotRequest) returns (SnapshotList) {
```

Comment:
Normally proto `rpc` argument and return values use convention of `methodnameRequest` `methodnameResponse`. Any reason we chose to not follow it?

Reply:
Some changes are pending on this proto...

the methods will be changed to
```
UploadDescriptor
DownloadDescriptor
ListSnapshots
```